File: WebApp/res/js/Indigox/UUM/Application/HREmployee.js
define("Indigox/UUM/Application/HREmployee",
    [
        "Indigox.Web.JsLib.Models.RecordManager"
    ],
function (
    RecordManager
) {
    RecordManager.getInstance().register('HREmployee', {
        columns: [
            { name: 'ID', text: '编号', type: String },
            { name: 'ParentID', text: '部门编号', type: String },
            { name: 'ParentName', text: '部门名称', type: String },
            { name: 'Name', text: '名称', type: String },
            { name: 'AccountName', text: '帐号', type: String },
            { name: 'DisplayName', text: '显示名称', type: String },
            { name: 'Tel', text: '电话', type: String },
            { name: 'Mobile', text: '手机', type: String },
            { name: 'IdCard', text: '身份证号', type: String },
            { name: 'Fax', text: '传真', type: String },
            { name: 'Email', text: '邮件', type: String },
            { name: 'Title', text: '头衔', type: String },
            { name: 'Enabled', text: '是否启用', type: String },
            { name: 'State', text: '状态', type: String },
            { name: 'Synchronized', text: '是否已同步', type: String },
            { name: 'Description', text: '描述', type: String },
            { name: 'MailDatabase', text: '邮箱数据库', type: String },
            { name: 'HasPolyphone', text: '含义多音字', type: String },
            { name: 'QuitDate', text: '离职日期', type: String }
        ],
        primaryKey: ['ID']
    });
});

File: Sources/Indigox.UUM.HR/Model/HREmployee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Indigox.UUM.HR.Model
{
    public enum HRState
    {
        Created, Changed, Deleted, Enabled, Disabled, Undefined
    }
    public class HREmployee
    {
        public string ID { get; set; }
        public string ParentID { get; set; }
        public string ParentName { get; set; }
        public string Name { get; set; }
        public string IdCard { get; set; }
        public string AccountName { get; set; }
        public string DisplayName { get; set; }
        public string Tel { get; set; }
        public string Mobile { get; set; }
        public string Fax { get; set; }
        public string Email { get; set; }
        public string Title { get; set; }
        public bool Enabled { get; set; }
        public bool EmployeeFlag { get; set; }
        public HRState State { get; set; }
        public bool Synchronized { get; set; }
        public DateTime ModifyTime { get; set; }
        public string Portrait { get; set; }
        public string Description { get; set; }
        public bool HasPolyphone { get; set; }
        public string MailDatabase { get; set; }
        public double OrderNum { get; set; }
        public IDictionary<string, string> ExtendProperties { get; set; } = new Dictionary<string, string>();
        public IList<HROrganizationalRole> OrganizationalRoles { get; set; }

        private DateTime quitDate = DateTime.Parse("2099-01-01");

        public DateTime QuitDate
        {
            get { return quitDate; }
            set { quitDate = value; }
        }
        public IDictionary<string, string> CloneExtendProperties()
        {
            Dictionary<string, string> ret = new Dictionary<string, string>(this.ExtendProperties.Count);
            foreach (KeyValuePair<string, string> entry in this.ExtendProperties)
            {
                if (entry.Value != null)
                {
                    ret.Add(entry.Key, (string)entry.Value.Clone());
                }
                else
                {
                    ret.Add(entry.Key, "");
                }
            }
            return ret;
        }
    }
}
